tests/CMakeLists.txt

```diff
@@ -128,7 +128,9 @@ set(PYBIND11_TEST_FILES
     test_custom_type_casters
     test_custom_type_setup
     test_docstring_options
-    test_eigen
+    test_eigen_matrix
+    test_eigen_tensor
+    test_eigen_tensor_no_array.cpp
```

Comment:
Picking this as a random point to comment on the name:

The "no_array" name means people have to learn to map "no_array" to EIGEN_AVOID_STL_ARRAY. To make the connection more obvious, I'd go with (and similarly everywhere else):

test_eigen_tensor_avoid_stl_array.cpp